util/git/git.go

```diff
@@ -43,20 +43,24 @@ func IsTruncatedCommitSHA(sha string) bool {
 
 // NormalizeGitURL normalizes a git URL for lookup and storage
 func NormalizeGitURL(repo string) string {
-	// preprocess
 	repo = strings.TrimSpace(repo)
-	repo = ensureSuffix(repo, ".git")
 	if IsSSHURL(repo) {
 		repo = ensurePrefix(repo, "ssh://")
 	}
-
-	// process
 	repoURL, err := url.Parse(repo)
 	if err != nil {
 		return ""
 	}
-
-	// postprocess
+	hostname := repoURL.Hostname()
+	if !strings.HasSuffix(hostname, "dev.azure.com") && !strings.HasSuffix(hostname, "visualstudio.com") {
```

Comment:
@jessesuen wondering if intent might be more clear factoring out into an `isAzureDomain()` function.